core/src/main/scala/caliban/wrappers/Wrappers.scala

```diff
@@ -0,0 +1,60 @@
+package caliban.wrappers
+
+import caliban.CalibanError.ExecutionError
+import caliban.Value.NullValue
+import caliban.wrappers.Wrapper.OverallWrapper
+import caliban.{ GraphQL, GraphQLResponse }
+import zio.ZIO
+import zio.clock.Clock
+import zio.console.{ putStrLn, Console }
+import zio.duration.Duration
+
+object Wrappers {
+
+  /**
+   * Attaches to the given GraphQL API definition a wrapper that prints slow queries.
+   * @param duration threshold above which queries are considered slow
+   * @param api a GraphQL API definition
+   */
+  def printSlowQueries[R <: Console with Clock](duration: Duration)(api: GraphQL[R]): GraphQL[R] =
+    api.withWrapper(printSlowQueriesWrapper(duration))
+
+  /**
+   * Returns a wrapper that prints slow queries
+   * @param duration threshold above which queries are considered slow
+   */
+  def printSlowQueriesWrapper(duration: Duration): OverallWrapper[Console with Clock] =
+    OverallWrapper {
+      case (io, query) =>
+        io.timed.flatMap {
+          case (time, res) =>
+            ZIO.when(time > duration)(putStrLn(s"Slow query took ${time.render}:\n$query")).as(res)
+        }
+    }
```

Comment:
Could also make one more to do generic effects on slow query `onSlowQueriesWrapper`:

```scala
  def onSlowQueriesWrapper(duration: Duration)(f: (Duration, String) => UIO[Any]): OverallWrapper[Console with Clock] =
    OverallWrapper {
      case (io, query) =>
        io.timed.flatMap {
          case (time, res) =>
            ZIO.when(time > duration)(f(time, res)).as(res)
        }
    }

```

which would allow for easy creation of logging mechanisms.

Reply:
Good idea, and I can implement `printSlowQueries` using it.